cachelib/shm/PosixShmSegment.cpp

```diff
@@ -312,13 +314,52 @@ void* PosixShmSegment::mapAddress(void* addr) const {
     util::throwSystemError(EINVAL, "Address already mapped");
   }
   XDCHECK(retAddr == addr || addr == nullptr);
+  memBind(addr);
   return retAddr;
 }
 
 void PosixShmSegment::unMap(void* addr) const {
   detail::munmapImpl(addr, getSize());
 }
 
+static void forcePageAllocation(void* addr, size_t size, size_t pageSize) {
+  for(volatile char* curAddr = (char*)addr; curAddr < (char*)addr+size; curAddr += pageSize) {
```

Comment:
```suggestion
  for (volatile char* curAddr = (char*)addr; curAddr < (char*)addr+size; curAddr += pageSize) {
```

Reply:
please use `reinterpret_cast` instead of c-style casting.

Reply:
Done